src/main/java/com/commercetools/sync/commons/utils/CustomUpdateActionUtils.java

```diff
@@ -261,24 +398,40 @@
+     * @param <T>                    the type of the old {@link Resource} which has the custom fields.
+     * @param <U>                    the type of the resource in which the update actions will be applied on.
+     * @param oldCustomFields        the old resource's custom fields map of JSON values.
+     * @param newCustomFields        the new resources's custom fields map of JSON values.
+     * @param resource               the resource that the custom fields are on. It is used to identify the type of
+     *                               the resource, to call the corresponding update actions.
+     * @param containerResourceClass the class of the container resource which will be updated.
+     * @param variantId              optional field representing the variant id in case the oldResource is an asset.
+     * @param resourceIdGetter       a function used to get the id of the resource being updated.
+     * @param resourceTypeIdGetter   a function used to get the Type id of the resource being updated.
+     * @param updateIdGetter         a function used to get the id/key needed for updating the resource that has the
+     *                               custom fields.
+     * @param syncOptions            responsible for supplying the sync options to the sync utility method.
      * @return a list that contains all the update actions needed, otherwise an
-     *      empty list if no update actions are needed.
+     *         empty list if no update actions are needed.
      */
     @Nonnull
-    static <T extends Custom & Resource<T>> List<UpdateAction<T>> buildRemovedCustomFieldsUpdateActions(
+    static <T extends Custom, U extends Resource<U>> List<UpdateAction<U>> buildRemovedCustomFieldsUpdateActions(
         @Nonnull final Map<String, JsonNode> oldCustomFields,
         @Nonnull final Map<String, JsonNode> newCustomFields,
         @Nonnull final T resource,
+        @Nullable final Class<U> containerResourceClass,
+        @Nullable final Integer variantId,
+        @Nonnull final Function<T, String> resourceIdGetter,
+        @Nonnull final Function<T, String> resourceTypeIdGetter,
+        @Nonnull final Function<T, String> updateIdGetter,
         @Nonnull final BaseSyncOptions syncOptions) {
         return oldCustomFields.keySet().stream()
                               .filter(oldCustomFieldsName -> Objects.isNull(newCustomFields.get(oldCustomFieldsName)))
                               .map(oldCustomFieldsName -> buildTypedSetCustomFieldUpdateAction(
-                                  oldCustomFieldsName, null, resource, syncOptions).orElse(null))
+                                  oldCustomFieldsName,
+                                  null, resource, containerResourceClass, variantId, resourceIdGetter,
+                                  resourceTypeIdGetter, updateIdGetter,
+                                  syncOptions).orElse(null))
                               .filter(Objects::nonNull)
```

Comment:
One more benefit to use stream abstraction design over optionals+streams:
in this case we could omit `orElse(null)` followed by `filter(Objects::nonNull)` just using only `Stream.flatMap()` 
(not but not to fix in this PR)